### PR TITLE
Fix python-gevent missing dependencies, broken tests, and CVE-2024-24806

### PR DESCRIPTION
**Pull request by nisamson**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [x] Ready to merge

---

###### Summary 

This PR makes several fixes to python-gevent. Previously, ptests were not running because they could not possibly run on 3.12. Additionally, the spec was not using the pyproject build process. Finally, the package was previously upgraded, but one of its core dependencies was not added to Azure Linux.

Several tests are either inherently flaky or were reliant on networking configuration assumptions which are not valid in our build process. These have been disabled via patch.

Finally, a patch was added for CVE-2024-24806 in the embedded version of libuv included in the project source.

###### Change Log  

- Add patch for CVE-2024-24806
- Update python-gevent build process and dependencies to enable testing.
- Patch out tests which are inherently flaky or which rely on specific configurations for networking.
- Add missing zope.event dependency to core specs.

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2024-24806

###### Test Methodology

- Pipeline build id: 599044

### CODE REVIEW

File: SPECS/python-gevent/skip-irrelevant-tests.patch [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ 3 files changed, 5 insertions(+), 63 deletions(-)
+
+diff --git a/src/gevent/tests/test__core_timer.py b/src/gevent/tests/test__core_timer.py
+index 97caa754..8e0b691f 100644
+--- a/src/gevent/tests/test__core_timer.py
++++ b/src/gevent/tests/test__core_timer.py
+@@ -95,63 +95,8 @@ class TestTimerResolution(Test):
+     # 0.02 or higher.
+     @reraises_flaky_timeout(AssertionError)
+     def test_resolution(self): # pylint:disable=too-many-locals
```
Comment (jslobodzian) [inline SPECS/python-gevent/skip-irrelevant-tests.patch:19]: Is there are more central place that we could patch instead?  Somewhere we can just remove the invocatio of the test?

Comment (nisamson) [reply SPECS/python-gevent/skip-irrelevant-tests.patch:19]: Not trivially; the tests are [discovered](https://github.com/gevent/gevent/blob/master/tox.ini) through a single invocation. Additionally, some of the tests are dynamically generated/invoked, meaning the only way to disable them properly is to either disable the entire suite (which is overly broad) or to bypass them in the source, like in this patch.

Comment (nisamson) [reply SPECS/python-gevent/skip-irrelevant-tests.patch:19]: Here's an example of [the generated tests.](https://github.com/microsoft/azurelinux/pull/9549/files/4acfb88b2c7435aa48ee8fc3bd9501bedaa14f87#diff-cdd57ffc0117ea266a8b21e69572744a26b095d1b9e6fd37044027f5a4e0203eR90-R100)

File: SPECS/python-gevent/python-gevent.spec [hunk 2]
```diff
@@ -42,23 +51,31 @@ Features include:
...
+%tox
 
-%files -n python3-gevent
+%files -n python3-gevent -f %{pyproject_files}
 %defattr(-,root,root,-)
 %license LICENSE
-%{python3_sitelib}/*
 
 %changelog
+* Mon Jul 01 2024 Nick Samson <nisamson@microsoft.com> -23.9.1-3
```
Comment (anphel31) [inline SPECS/python-gevent/python-gevent.spec:74]: missing space: `-23.9.1-3`

**Review by jslobodzian — APPROVED** [review]

**Pull request merged**